### PR TITLE
Experimental -- v4 provider

### PR DESCRIPTION
**Pull request by calina-c**
List of PR changes:
* replace old ECIES_PRIVATE_KEY with PRIVATE_KEY. The purpose of the PRIVATE_KEY is not to decrypt, but to identify the Aquarius version when asking for decrypt from Provider
* temporarily added the artifacts folder and artifacts/install.sh script until we have an ocean-contracts release
* where functions depended on the old DDO structure, replace with the new one, e.g. instead of using `get_metadata_from_services`, we can now use the `"metadata"` key directly. Other mischellaneous changes related to the structure, as needed (not all listed below):
	* replace the date format with ISO Format
* remove the encrypt_ddo and encrypt_as_hex endpoints and the encrypt_data function
* repurpose the decryptor class to contain utilitaries for “talking to” Provider
* remove metadata contract from events monitor, since there are now multiple contracts
* reconfigure the event logs retrieval in events monitor to filter by topics for MetadataCreated and MetadataUpdated events, regardless of contract
* replace some logic in processors.py to handle the new structure/event arguments
* remove eciespy from dependencies (no longer needed)

Changes in tests:
* remove deployment of datatoken contract
* remove rawddos (pertaining to old processor tests)
* repurpose send_create_update_tx function to use setMetaData
* use v4 samples
* add unit and functional tests where needed

TODOs:
- [x] check_no_urls_in_files -> adapt
- [x] remove curation :D
- [x] check all arguments for events processors are used
- [ ] clear v3 samples ?
- [ ] add "real" metadata start block: should be taken from barge, but it is not currently written in the address file

### CODE REVIEW

**Comment by calina-c** [comment]
Screenshot 2021-11-04 at 14 49 40

File: aquarius/events/processors.py [hunk 1]
```diff
@@ -132,14 +133,31 @@ def make_record(self, data):
...
+        )
+
+        if not self.check_document_hash(asset):
+            return False
+
+        self.did = asset["id"]
         did, sender_address = self.did, self.sender_address
         logger.info(
-            f"Process new DDO, did from event log:{did}, sender:{sender_address}, flags: {self.flags}, block {self.block}, contract: {self.contract_address}, txid: {self.txid}, chainId: {self._chain_id}"
+            f"Process new DDO, did from event log:{did}, block {self.block}, contract: {self.event.address}, txid: {self.txid}, chainId: {self._chain_id}"
```
Comment (DMats) [inline aquarius/events/processors.py:160]: Please consider wrapping this line for readability.

File: aquarius/events/processors.py [hunk 2]
```diff
@@ -96,8 +98,8 @@ def make_record(self, data):
         _record["event"] = {
             "txid": self.txid,
             "blockNo": self.block,
-            "from": self.sender_address,
-            "contract": self.contract_address,
+            "from": self.event.address,
```
Comment (DMats) [inline aquarius/events/processors.py:101]: `event.address` is not equivalent to the `"from"` address.  I recommend using `self.sender_address`.

File: aquarius/events/processors.py [hunk 3]
```diff
@@ -229,42 +242,61 @@ def make_record(self, data, asset):
...
+                self.sender_address,
                 self._es_instance,
                 self._web3,
-                self._ecies_account,
                 self.allowed_publishers,
                 self.purgatory,
                 self._chain_id,
             )
             event_processor.process()
             return False
```
Comment (DMats) [inline aquarius/events/processors.py:293]: Why return `False` here? Shouldn't it return the result of the line above?

File: aquarius/app/util.py [hunk 4]
```diff
@@ -70,21 +68,14 @@ def get_metadata_from_services(services):
...
-
-
 def init_new_ddo(data, timestamp):
     _record = copy.deepcopy(data)
     _record["created"] = format_timestamp(
         datetime.fromtimestamp(timestamp).strftime(DATETIME_FORMAT)
     )
     _record["updated"] = _record["created"]
 
+    # TODO: still needed?
```
Comment (DMats) [inline aquarius/app/util.py:78]: The [v4 DDO spec](https://github.com/oceanprotocol/docs/blob/feature/ddo_v4/content/concepts/did-ddo.md) doesn't mention `accessWhiteList` so I think we should remove this and all other mentions of `accessWhiteList`.

Just to be sure, I searched through historical commits and issues.

Issue https://github.com/oceanprotocol/multi-repo-issue/issues/50 was going to use the `allowWhiteList` property, but it was closed without changes.
This code was added in PR #266 but even at that time it seems to be a vestige of an older design.  In the same PR, multiple unit tests relating to `accessWhiteList` were removed. 
* Added: https://github.com/oceanprotocol/aquarius/pull/266/files#diff-a16473cce6b6252e953d1ae0ac253984c03eff6fa90ede50ed02cdda6ed4e9d7R96-R102
* Removed unit tests: https://github.com/oceanprotocol/aquarius/pull/266/files#diff-3451cc474d5698222168f66a293339a6bd296ee2c56f56409866bbae99917a79L409-L482

**Pull request merged**